### ISSUE TITLE
Cannot cast ufunc subtract output from ...

### ISSUE THREAD

**Issue by karlind** [mahyarnajibi/SNIPER#27]
Hi, I got a problem which really confused me for several days, and I cannot solve. If you may help, that will be great.
The error message shows bellow:
```
loading annotations into memory...
Done (t=4.05s)
creating index...
index created!
num_images 40504
COCO_val2014 gt roidb loaded from ./data/cache/COCO_val2014_gt_roidb.pkl
filtered 367 roidb entries: 40504 -> 40137
add bounding box regression targets
bbox target means:
[[0. 0. 0. 0.]
 [0. 0. 0. 0.]]
[0. 0. 0. 0.]
bbox target stdevs:
[[0.1 0.1 0.2 0.2]
 [0.1 0.1 0.2 0.2]]
[0.1 0.1 0.2 0.2]
Creating Iterator with 40137 Images
Total number of extracted chips: 155974
Done!
Traceback (most recent call last):
  File "/home/arias/Projects/Python/SNIPER/main_train.py", line 74, in <module>
    threads=config.TRAIN.NUM_THREAD, pad_rois_to=400)
  File "lib/iterators/MNIteratorE2E.py", line 31, in __init__
    super(MNIteratorE2E, self).__init__(roidb, config, batch_size, threads, nGPUs, pad_rois_to, False)
  File "lib/iterators/MNIteratorBase.py", line 33, in __init__
    self.get_batch()
  File "lib/iterators/MNIteratorE2E.py", line 99, in get_batch
    self.batch = self._get_batch()
  File "lib/iterators/MNIteratorE2E.py", line 171, in _get_batch
    all_labels = self.pool.map(self.anchor_worker.worker, worker_data)
  File "/home/arias/anaconda3/envs/python2/lib/python2.7/multiprocessing/pool.py", line 253, in map
    return self.map_async(func, iterable, chunksize).get()
  File "/home/arias/anaconda3/envs/python2/lib/python2.7/multiprocessing/pool.py", line 572, in get
    raise self._value
TypeError: Cannot cast ufunc subtract output from dtype('float64') to dtype('uint16') with casting rule 'same_kind'
```

**Comment by karlind** [comment]
System: Ubuntu 16.04
Python version: 2.7(Anaconda)

And I set `only_gt=True`, `flip=False`. When I keep `only_gt=False`, my memory decrease quickly from 16GB to 0, so I have to do that as a temp solution.

**Comment by bharatsingh430** [comment]
you can try to reduce the number of processes and increase the CHIPS_DB_PARTS in the config file, but 16GB for coco might still be a bit less

**Comment by mahyarnajibi** [comment]
Hi @karlind ,
The cast error should be addressed in commit 8711340. Instead of directly setting ```only_gt``` you can set the ```USE_NEG_CHIPS``` to False which disables the need for using the proposals and consequently the negative chip mining. However, 16GB still might not be sufficient in your case. You can reduce the ```NUM_PROCESS``` and increase ```CHIPS_DB_PARTS``` if you still face memory issues to see if it helps.

**Comment by karlind** [comment]
@mahyarnajibi Thanks for your quick response. Your response really help. And now I run in coco dataset.
@bharatsingh430 You are right, 16GB is a bit less, so I cut out a subset of coco, and it works. Thank you.
